src/python/pants/backend/go/module_integration_test.py

```diff
@@ -55,3 +65,22 @@ def test_resolve_go_module(rule_runner: RuleRunner) -> None:
         if module_descriptor.module_path == "github.com/golang/protobuf":
             found_protobuf_module = True
     assert found_protobuf_module
+
+
+def test_download_external_module(rule_runner: RuleRunner) -> None:
+    downloaded_module = rule_runner.request(
+        DownloadedExternalModule,
+        [
+            DownloadExternalModuleRequest(
+                path="github.com/google/uuid",
+                version="v1.3.0",
+            )
+        ],
+    )
+    digest_contents = rule_runner.request(DigestContents, [downloaded_module.digest])
+    found_uuid_go_file = False
+    for file_content in digest_contents:
+        if file_content.path == "uuid.go":
+            found_uuid_go_file = True
+            break
+    assert found_uuid_go_file
```

Comment:
Maybe worth asserting that `DownloadExternalModuleRequest.{path,version}` are the same as the returned `downloaded_module`.

Reply:
Done.